### PR TITLE
Add animated GIF splash support

### PR DESCRIPTION
**Pull request by Epicpkmn11**
This adds animated GIF splash support and cleans up the main file a bit (it looks like more that it really is as I moved a lot of stuff around)

GIF details:
- Most GIFs up to 256x192 and less than around 10MB (more than you should ever need) between both images should work
  - Smaller GIFs will be centered
- GIFs up to 1MB will be decompress while loading and should all display at full speed, GIFs larger than that may play a bit slowly since their decompressed while displaying
- If both screens loop indefinitely (or aren't GIFs) then they will be shown for 3 seconds (same as before)
- Otherwise they will be shown until both GIFs have completed their set amount of loops, if one finishes first it will wait on the last frame
- The user input flag is supported and the GIF will pause on that frame until any button pressed
  - *Did you even know GIFs can pause until input? Like nothing supports it ;P*
  - Edit: As of #17 it will wait until any button is pressed ***or*** the delay on that frame runs out, so if you set a 1 minute delay it will wait 1 minute or until a button is pressed
- Interlaced and text GIFs are currently ***not*** supported
  - Text because its so rare I couldn't even find a tool to make one
  - Interlaced because it doesn't seem to provide any major advantages (all fields seem to be supposed to be shown at once?) and would be kinda annoying to do I think ;P
- TL;DR all vaguely normal GIFs should work (and even some weirder things) ;P

Misc:
- BMPs are still supported and now any up to 256x192 will work
- It will check for GIF, then a BMP, then if neither are found it will use the default splash for that screen
- Switched the GRIT images to be bitmaps instead of map and tile since that's easier to display
- A couple changes to the Makefiles and .gitignore
- Rearranged main.cpp a bit, but mostly only the graphics related things have changed
- Switched to tonccpy intead of vramcpy_ui

Todo:
- [x] Make sure it still actually works
- [x] Maybe try to improve speed

### CODE REVIEW

**Comment by Epicpkmn11** [comment]
Alright, I made it decompress GIFs up to 1MB while loading instead of while displaying so now it can display all GIFs I've tried (up to 1MB) at full speed so I think this is probably good.

Here's the GIFs I tested with:
- [Newton's Cradle](https://user-images.githubusercontent.com/41608708/90265549-57b0b000-de18-11ea-9cb6-7625fdbb55a6.gif), has small animated areas so this worked fine even while decompressing while displaying
- [Spinning earth](https://user-images.githubusercontent.com/41608708/90265725-96df0100-de18-11ea-91b8-07f3a5b7f407.gif), only the earth is animated so this is also pretty light
- [Earth ring thing](https://user-images.githubusercontent.com/41608708/90265662-83cc3100-de18-11ea-94cc-543265715d3b.gif), the whole image is animated pretty fast so this was slow before
- [Cardcaptor Sakura OP1](https://user-images.githubusercontent.com/41608708/90265810-bb3add80-de18-11ea-99ab-52dadceb8682.gif), this is about as big as you can use, since its way to big to fit in RAM decompressed it still plays a bit slow
- [iQue top splash](https://user-images.githubusercontent.com/41608708/90294764-d293be00-de4c-11ea-8be1-4a8dcec8de45.gif), from TWiLight, this waits for user input on the last frame
- [Japanese H&S message](https://user-images.githubusercontent.com/41608708/90294798-eccd9c00-de4c-11ea-990c-33fef2d863cd.gif), from TWiLight, loops forever
